packages/experiments/src/utilities/BaseComponentMin.ts

```diff
@@ -0,0 +1,178 @@
+import * as React from 'react';
+import { IStateComponentProps } from '../Foundation';
+import { IDisposable, IRefObject } from '../Utilities';
+
+/**
+ * BaseProps interface.
+ *
+ * @public
+ */
+// tslint:disable-next-line:no-any
+export interface IBaseProps<T = any> {
+  componentRef?: IRefObject<T>;
+}
+
+/**
+ * BaseComponent class, which provides basic helpers for all components.
+ *
+ * @public
+ */
+export class BaseComponentMin<TComponentProps extends IBaseProps = {}, TViewProps = {}, TState = {}> extends React.Component<
+  IStateComponentProps<TComponentProps, TViewProps>,
+  TState
```

Comment:
@JasonGore I have some doubts about a change I made here to make it work with the separation between state and view but I don't know if it was the correct thing to do.

Originally, this was:

```ts
export class BaseComponent<TProps extends IBaseProps = {}, TState = {}> extends React.Component<TProps, TState> {
```

What do you think?

Reply:
This is at a lower level so BaseComponentMin should not have to know anything about state components or state vs. view classes, and shouldn't be using anything from `IStateComponentProps`. The only thing BaseComponentMin should be concerned about is that props has `componentRef`, which is covered by `extends IBaseProps`. Line 21 should be able to be just `TComponentProps`.